src/main/java/org/graylog/integrations/pagerduty/client/PagerDutyClient.java

```diff
@@ -0,0 +1,116 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.integrations.pagerduty.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.graylog.integrations.pagerduty.dto.PagerDutyResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+/**
+ * The Pager Duty REST client implementation class compatible with events V2. For more information
+ * about the event structure please see
+ * <a href="https://v2.developer.pagerduty.com/v2/docs/send-an-event-events-api-v2">the api</a>.
+ *
+ * This class is heavily based on the work committed by Jochen, James, Dennis, Padma, and Edgar
+ * <a href="https://github.com/graylog-labs/graylog-plugin-pagerduty/">here</a>.
+ *
+ * @author Jochen Schalanda
+ * @author James Carr
+ * @author Dennis Oelkers
+ * @author Padma Liyanage
+ * @author Edgar Molina
+ */
+public class PagerDutyClient {
+    private static final Logger LOG = LoggerFactory.getLogger(PagerDutyClient.class);
+
+    @VisibleForTesting
+    static final String API_URL = "https://events.pagerduty.com/v2/enqueue";
+
+    private final OkHttpClient httpClient;
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public PagerDutyClient(final OkHttpClient httpClient,
+                           final ObjectMapper objectMapper) {
+        this.httpClient = httpClient;
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * This method POSTs a message to PagerDuty's events enqueue API.
+     *
+     * @param payloadString JSON representation of a valid PagerDuty change event or alert event
+     * @return PagerDutyResponse object
+     * @throws TemporaryPagerDutyClientException when a retryable error is encountered
+     * @throws PermanentPagerDutyClientException when a non-retriable error is encountered
+     */
+    public PagerDutyResponse enqueue(String payloadString)
+            throws TemporaryPagerDutyClientException, PermanentPagerDutyClientException {
+        final Request request = new Request.Builder()
+                .url(API_URL)
+                .post(RequestBody.create(MediaType.parse(APPLICATION_JSON), payloadString))
+                .build();
+
+        LOG.debug("Triggering event in PagerDuty with POST payload: {}", payloadString);
+
+        try (final Response response = httpClient.newCall(request).execute()) {
+            if (!response.isSuccessful()) {
```

Comment:
please add `log.debug` to print essential elements like `response headers` ,`status code`, `response size`, total time taken to get the response back etc.

Reply:
Will do.